repos/system_upgrade/el7toel8/actors/persistentnetnamesconfig/tests/test_persistentnetnamesconfig.py

```diff
@@ -114,3 +137,32 @@ def test_renamed_swap_eth(monkeypatch, current_actor_context):
         elif interface.rhel7_name == 'eth3':
             assert interface.rhel8_name == 'eth0'
     assert not initrd_files.files
+
+
+def test_bz_1899455_crash_iface(monkeypatch, current_actor_context, adjust_cwd):
+    """
+    Cover situation when network device is discovered on the src sys but not
+    inside the upgrade environment.
+
+    This typically happens when the network device needs specific drivers which
+    are not present inside the upgrade initramfs. Usually it points to a missing
+    actors that should influence the upgrade initramfs in a way the drivers are
+    installed. In this situation, only correct thing we can do in this actor
+    is print warning / report that we couldn't located particular devices so
+    we cannot handle interface names related to this devices.
+    """
+    with open(os.path.join(CUR_DIR, 'files/crashed_ifaces.json')) as fp:
+        json_msgs = json.load(fp)
+    msgs = [
+        PersistentNetNamesFacts.create(json_msgs["PersistentNetNamesFacts"]),
+        PersistentNetNamesFactsInitramfs.create(json_msgs["PersistentNetNamesFactsInitramfs"]),
+    ]
+    monkeypatch.setattr(persistentnetnamesconfig, 'generate_link_file', generate_link_file_mocked)
+    monkeypatch.setattr(persistentnetnamesconfig.api, 'current_actor', CurrentActorMocked(msgs=msgs))
+    monkeypatch.setattr(persistentnetnamesconfig.api, 'current_logger', logger_mocked())
+    monkeypatch.setattr(persistentnetnamesconfig.api, 'produce', produce_mocked())
+    persistentnetnamesconfig.process()
+
+    for prod_models in [RenamedInterfaces, InitrdIncludes]:
+        any(isinstance(i, prod_models) for i in persistentnetnamesconfig.api.produce.model_instances)
+    assert any(['Some network devices' in x for x in persistentnetnamesconfig.api.current_logger.warnmsg])
```

Comment:
Just wondering if logging a warning is enough in that case. Maybe we should report it as well?

Reply:
I was thinking the same thing, but we should fix it first. We can make a follow up on that. We need to decide on what to tell the user what they should do in that case.

Reply:
I guess it's enough now regarding the rest of actor (where the warning is used only as well in cases we cannot handle). But in future I think we should generate post-upgrade reports as well.

Reply:
Sure, I have no problem with that. Just please leave smth like `TODO add proper reporting later`

Reply:
the TODO is added.